lib/ngx/re.lua

```diff
@@ -121,24 +130,21 @@ function _M.split(subj, regex, opts, ctx, max, res)
         return res
     end
 
-    if regex == "" then
-        local pos = ctx.pos
-        local last = len
-        if max > 0 then
-            last = math_min(len, pos + max - 1)
-        end
+    local res_idx = 0
+    local sub_idx = ctx.pos
 
-        local res_idx = 1
-        while pos < last do
-            res[res_idx] = sub(subj, pos, pos)
-            res_idx = res_idx + 1
-            pos = pos + 1
-        end
+    local start_regex = find(regex, "^", nil, true) ~= nil
```

Comment:
@thibaultcha Well, here we should follow our own semantics instead of perl's:

```console
$ resty -e 'print((ngx.re.match("a\nb", "^b", "")) and "true" or "false")'
false

$ resty -e 'print((ngx.re.match("a\nb", "^b", "m")) and "true" or "false")'
true
```